crates/wit-component/src/encoding/wit/mod.rs

```diff
@@ -43,12 +43,12 @@ pub fn encode(use_v2: Option<bool>, resolve: &Resolve, package: PackageId) -> Re
 pub fn encode_component(
     use_v2: Option<bool>,
     resolve: &Resolve,
-    package: PackageId,
+    packages: &[PackageId],
 ) -> Result<ComponentBuilder> {
     if use_v2.unwrap_or_else(use_v2_encoding) {
-        v2::encode_component(resolve, package)
+        v2::encode_component(resolve, packages)
     } else {
-        v1::encode_component(resolve, package)
+        v1::encode_component(resolve, packages[0])
```

Comment:
This should also update to generate an error if `packages.len() > 0` since otherwise it's losing information